src/ingameWindows/iwMapGenerator.cpp

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2005-2010 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Siedler II.5 RTTR.
+//
+// Siedler II.5 RTTR is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Siedler II.5 RTTR is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Siedler II.5 RTTR. If not, see <http://www.gnu.org/licenses/>.
+
+#include "defines.h" // IWYU pragma: keep
+#include "iwMapGenerator.h"
+
+#include "Loader.h"
+#include "controls/ctrlOptionGroup.h"
+#include "controls/ctrlComboBox.h"
+#include "controls/ctrlProgress.h"
+
+#include "gameData/const_gui_ids.h"
+#include "gameData/MaxPlayers.h"
+#include "helpers/containerUtils.h"
+#include "libutil/src/colors.h"
+
+#include <boost/format.hpp>
+#include <string>
+
+enum
+{
+    CTRL_PLAYER_NUMBER = 2,
```

Comment:
You can use a placeholder. Like start with `CTRL_LAST_ID = 10` and then all your named ctrls afterwards. Then you can use id 1-10 for the unnamed ones and the others will be named with ids starting at 11. This way you only need to specify one value and if you add more unnamed values you can simply change the placeholder